src/newgrf_station.h

```diff
@@ -178,7 +178,7 @@ struct StationSpec : NewGRFSpecBase<StationClassID> {
 	 * These can be sparsely populated, and the upper limit is not defined but
 	 * limited to 255.
 	 */
-	std::vector<std::vector<std::vector<uint8_t>>> layouts;
+	std::unordered_map<uint16_t, std::vector<uint8_t>> layouts;
```

Comment:
wonder if it would be overkill to add some getter/setters to make it clear what the key is

(also the doc comment could do with an update)

Reply:
Added a function to determine the key. Mostly lots of lines due to doxygen.